enhancements/installer/install_ca_management.md

```diff
@@ -0,0 +1,161 @@
+associated to registry hostnames.  Further complicating the situation is that the field is simply 
+named `additionalTrustBundle` so the user intent is not clear, making it difficult for specific 
+components to know whether they should be consuming this value or not.
+
+The most direct consequences of this are that today if a customer does a disconnected install, 
+their imagestream imports from the mirror registry will fail.  In particular this means the 
+oauth-proxy imagestream, which is consumed by many OLM operators, will fail and thus those 
+operators will fail to function, until/unless the admin explicitly configures the CA in the image 
+config resource and then re-imports the imagestream.
+
+While this could theoretically be solved by the imagestream component trusting the additional CA (
+for all hosts?) explicitly, it would make more sense if we used the information the admin is 
+providing us at install time to correctly configure the cluster as if they had done it themselves, 
+rather than introducing additional sources of (potentially conflicting) configuration 
+information.
+
+## Motivation
+
+See summary above, motivation is to make the install experience clearer with less potential for misconfiguration and conflicting settings.
+
+### Goals
+
+- Post install the cluster can not only pull images from an alternate registry, but imagestreams can import from it, builds can pull from it, and the internal registry can perform pullthrough against it
+- Post install the registry CAs provided by the admin are populated into the configuration api that is intended for this purpose
+
+### Non-Goals
+
+- Refactoring the additionalTrustBundle field into many unique fields for different CA usages
+- Addressing the current proxy CA behavior in which the additionalTrustBundle is sometimes populated into a proxy configuration and other times not (depending on whether a proxy host is supplied)
+
+
+## Proposal
+
+The installer should populate the 
+[image config resource](https://github.com/openshift/api/blob/master/config/v1/types_image.go) in the same way the 
+[product documentation](https://docs.openshift.com/container-platform/4.5/openshift_images/image-configuration.html#images-configuration-cas_image-configuration) 
+recommends.  The installer 
+can use the value of the additionalTrustBundle field to populate this api.  Similar to how the 
+value is used for the proxy config when a proxy host is supplied, the value can be used to 
+populate the image config resource when a mirror registry is in use.
+
+In addition, the downstream components that potentially communicate with the mirror registry should add logic which confirms they can successfully access the mirror registry.  Specifically, for any registry defined in the image content source policy, the components should be able to access the registry using the supplied CAs.  If they cannot, they should mark themselves degraded.
```

Comment:
Should the openshift-controller-manager-operator be responsible for doing this? It seems a bit harsh to mark ocm degraded when the result of a bad configuration is that builds fail.

Reply:
I think rather than having every component that consumes the registry CAs perform this check, it may make more sense to have a single component that checks the config and reports the degraded condition.  (Note that we need a degraded condition and not an alert because the intent is to block the install if the config is not correct).  

The registry operator (as the operator of the node-ca-daemon component) might be the most logical component to be responsible for the check and reporting itself degraded.  But that has the same problem as you note w/ OCM:  forcing the registry to mark itself degraded when it is mostly functional and just can't do pullthrough.

Another alternative is a brand new lightweight "config-checking" operator.  Which also feels icky.

Reply:
I don't think that all downstream components should check this. Should the downstream components report themselves as Degraded if the mirror registry is down? It feels wrong. The registry operator status is already strange, it reports the aggregated state of the registry, the image pruner, the node-ca daemonset, maybe this also might be reported by the registry operator (but afaik pull-through usually is not used with the mirror registry). But probably the registry operator shouldn't aggregate these states and we should have different operators instead. In this case the config-checking operator is a good idea.

Reply:
> Should the downstream components report themselves as Degraded if the mirror registry is down?

this is why an alert would probably be better(but that does not block the install), but components could check the error they are getting and only care about cert errors.

but yes i like the config-checking-operator better also.

Reply:
I would be interested to hear from @abhinavdahiya, @sdodson, and @crawford whether they would consider an alert sufficient (or other alternatives).  The main reason this section(requiring that the consuming components mark themselves degraded when this configuration is not correct) was because the installer team's position is that the installer should not take any action which is not required to complete the "create cluster" step.  Having an operator mark itself degraded if the config is not correct would be sufficient to block the completion of "create cluster" and thus make it reasonable to require the installer populate the configuration correctly.